modules/ximgproc/samples/deriche_demo.cpp

```diff
@@ -0,0 +1,110 @@
+ *  and / or other materials provided with the distribution.
+ *
+ *  * Neither the names of the copyright holders nor the names of the contributors
+ *  may be used to endorse or promote products derived from this software
+ *  without specific prior written permission.
+ *
+ *  This software is provided by the copyright holders and contributors "as is" and
+ *  any express or implied warranties, including, but not limited to, the implied
+ *  warranties of merchantability and fitness for a particular purpose are disclaimed.
+ *  In no event shall copyright holders or contributors be liable for any direct,
+ *  indirect, incidental, special, exemplary, or consequential damages
+ *  (including, but not limited to, procurement of substitute goods or services;
+ *  loss of use, data, or profits; or business interruption) however caused
+ *  and on any theory of liability, whether in contract, strict liability,
+ *  or tort(including negligence or otherwise) arising in any way out of
+ *  the use of this software, even if advised of the possibility of such damage.
+ */
+
+#include <opencv2/core.hpp>
+#include <opencv2/core/utility.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/ximgproc.hpp>
+#include "opencv2/ximgproc/deriche_filter.hpp"
+
+using namespace cv;
+using namespace cv::ximgproc;
+
+#include <iostream>
+using namespace std;
+
+int lowThreshold=20;
+int maxThreshold=20;
+int const max_lowThreshold = 500;
+int alDerive=100;
+int alMean=100;
+Mat dx, dy;
+UMat img;
+const char* window_name = "Gradient Modulus";
+
+static void DisplayImage(UMat x,string s)
+{
+	vector<Mat> sx;
+	split(x, sx);
+	vector<double> minVal(3), maxVal(3);
+	for (int i = 0; i < static_cast<int>(sx.size()); i++)
+	{
+		minMaxLoc(sx[i], &minVal[i], &maxVal[i]);
+	}
+	maxVal[0] = *max_element(maxVal.begin(), maxVal.end());
+	minVal[0] = *min_element(minVal.begin(), minVal.end());
+	Mat uc;
+	x.convertTo(uc, CV_8U,255/(maxVal[0]-minVal[0]),-255*minVal[0]/(maxVal[0]-minVal[0]));
+	imshow(s, uc);
+}
+
+
+/**
+ * @function DericheFilter
+ * @brief Trackbar callback
+ */
+static void DericheFilter(int, void*)
+{
+    UMat dst;
+    double d=alDerive/100.0,m=alMean/100.0;
+    UMat rx=  GradientDericheX(img,d,m);
+    UMat ry=  GradientDericheY(img,d,m);
+    DisplayImage(rx, "Gx");
+    DisplayImage(ry, "Gy");
+    add(rx.mul(rx),ry.mul(ry),dst);
+    sqrt(dst,dst);
+    DisplayImage(dst, window_name );
+}
+
+
+int main(int argc, char* argv[])
+{
+    if (argc==2)
+        imread(argv[1]).copyTo(img);
+    if (img.empty())
+    {
+        cout << "File not found or empty image\n";
+    }
+    imshow("Original",img);
+    namedWindow( window_name, WINDOW_AUTOSIZE );
+
+    /// Create a Trackbar for user to enter threshold
+    createTrackbar( "Derive:",window_name, &alDerive, 400, DericheFilter );
+    createTrackbar( "Mean:", window_name, &alMean, 400, DericheFilter );
+    DericheFilter(0,NULL);
+    waitKey();
```

Comment:
No I follow this https://github.com/opencv/opencv/blob/master/samples/cpp/tutorial_code/HighGUI/AddingImagesTrackbar.cpp